address-space-controller/src/main/java/io/enmasse/controller/NetworkPolicyController.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2017-2018, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.controller;
+
+import io.enmasse.address.model.*;
+import io.enmasse.admin.model.v1.InfraConfig;
+import io.enmasse.admin.model.v1.NetworkPolicy;
+import io.enmasse.config.AnnotationKeys;
+import io.enmasse.config.LabelKeys;
+import io.enmasse.k8s.api.SchemaProvider;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.networking.*;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class NetworkPolicyController implements Controller {
+    private final KubernetesClient kubernetesClient;
+    private final SchemaProvider schemaProvider;
+
+    public NetworkPolicyController(KubernetesClient kubernetesClient, SchemaProvider schemaProvider) {
+        this.kubernetesClient = kubernetesClient;
+        this.schemaProvider = schemaProvider;
+    }
+
+    @Override
+    public AddressSpace handle(AddressSpace addressSpace) throws Exception {
+        NetworkPolicy networkPolicy = null;
+        InfraConfig infraConfig = parseCurrentInfraConfig(addressSpace);
+        if (infraConfig != null) {
+            networkPolicy = infraConfig.getNetworkPolicy();
+        }
+
+        if (addressSpace.getNetworkPolicy() != null) {
+            networkPolicy = addressSpace.getNetworkPolicy();
+        }
+
+        String infraUuid = addressSpace.getAnnotation(AnnotationKeys.INFRA_UUID);
+        List<Service> services = kubernetesClient.services().withLabel(LabelKeys.INFRA_UUID, infraUuid).list().getItems();
```

Comment:
The service declaration should be moved to within the first branch of the if block.